pkg/config/store.go

```diff
@@ -78,6 +78,10 @@ type ChangeNotifier interface {
 	// The client should issue ReadChangeLog to see what has changed if the call is available.
 	// else it should re-read the store, perform diff and apply changes.
 	RegisterStoreChangeListener(s StoreListener)
+
+	// IsStoreChangeAvailable returns true only when the backend store is configured
+	// to emit change notifications.
+	IsStoreChangeAvailable() bool
```

Comment:
You just need `ChangeLogReader` to read and apply changes from the backing store.

Change notification is an optimization. So I suggest we simply make implementing `ChangeNotifier` interface optional. That way, if the a store implements it, we register to receive notifications.

We should be reading change log for new changes periodically anyway. If we get notified them we read it, but if the notification mechanism breaks down, we should still read it.

So we don't need the `IsStoreChangeAvailable()` method on that interface.